src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1302 @@
+    }
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select port.
+*
+* This test function selects HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 004@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_select_port(void)
+{
+    gTestID = 4;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select = 0; 
+    bool audmix = false;
+    dsVideoPlaneType_t videoplanetype = dsVideoPlane_MAX;
+    bool topmost = false;
+    uint8_t numInputPorts = 0;
+ 
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    UT_LOG_INFO(" \n================Please Select Inputs==================\n");
+    UT_LOG_INFO("\n*******Enter the port number to select*******\n");
+    
+    readInput(&select);
+    if(select <= 0 || select > (numInputPorts-1)) 
+    {
+       UT_LOG_ERROR("\nInvalid port selected\n");
+    }
+
+    port = (select - 1);
+
+    UT_LOG_INFO("\n*******Enter the audiomix to select*******\n\t"
+                     "\tAcceptable inputs are:\n\t"
+                     "\ttrue (i.e 1)\n\t"
+                     "\tfalse(i.e 0)\n\t"                    
+                "\n******************************************\n");
+
+    readInput(&select);
+    if(select < 0 || select > 1)
+    {
+       UT_LOG_ERROR("\nInvalid audmix selected\n");
+    }
+   
+    audmix = (bool)select;
+
+    UT_LOG_INFO("\n*******Enter the videoplanetype to select*******\n");
+    for(dsVideoPlaneType_t i = dsVideoPlane_PRIMARY ; i < dsVideoPlane_MAX ; i++)
+    {
+          UT_LOG_INFO("\n Available videoplane type: %s, videoplane number: %d \n",
+                 UT_Control_GetMapString(dsVideoPlaneType_mapTable,i),i);
+    }
+    UT_LOG_INFO("\n**************************************************\n");
+
+    readInput(&select);
+    if(select < 0 || select >= dsVideoPlane_MAX)
+    {
+       UT_LOG_ERROR("\nInvalid video plane type selected\n");
+    }
+
+    videoplanetype = (dsVideoPlaneType_t)select;
+
+    UT_LOG_INFO("\n*******Enter the topmost to select*******\n\t"
+                     "\tAcceptable inputs are:\n\t"
+                     "\ttrue (i.e 1)\n\t"
+                     "\tfalse(i.e 0)\n\t"
+                "\n******************************************\n");
+    
+    readInput(&select);
+    if(select < 0 || select > 1)
+    {
+       UT_LOG_ERROR("\nInvalid audmix selected\n");
+    }
+    
+    topmost = (bool)select;
+
+    ret = dsHdmiInSelectPort(port , audmix , videoplanetype , topmost);
+    UT_LOG_INFO("Result : dsHdmiInSelectPort: %s, portnumber: %d",
+                  UT_Control_GetMapString(dsHdmiInPort_mapTable, port), port);
+    assert(ret == dsERR_NONE);
```

Comment:
Better to add assert as MACRO in the top as

#define ASSERT assert
or
#define DS_ASSERT assert

This helps to replace if any changes occurs later

Reply:
done